Providers/Profile.swift

```diff
@@ -269,26 +269,16 @@ public class BrowserProfile: Profile {
         log.debug("Reopening profile.")
         isShutdown = false
         
-        if dbCreated {
-            db.reopenIfClosed()
-        }
-
-        if loginsDBCreated {
-            loginsDB.reopenIfClosed()
-        }
+        db.reopenIfClosed()
+        loginsDB.reopenIfClosed()
     }
 
     func shutdown() {
         log.debug("Shutting down profile.")
         isShutdown = true
 
-        if self.dbCreated {
-            db.forceClose()
-        }
-
-        if self.loginsDBCreated {
-            loginsDB.forceClose()
-        }
+        db.forceClose()
```

Comment:
The reason to have this flag here is that just calling `db.forceClose()` will _create_ `db` — doing a lot of disk access — in order to immediately shut it down!

This will typically be noticed more with the logins DB, which is more lazy.